libs/vkd3d/command.c

```diff
@@ -3400,6 +3546,11 @@ static bool d3d12_command_list_begin_render_pass(struct d3d12_command_list *list
     vk_render_pass = list->pso_render_pass;
     assert(vk_render_pass);
 
+    /* Emit deferred clears that we cannot clear inside the
+     * render pass, e.g. when the attachments to clear are
+     * not used by the render pass. */
+    d3d12_command_list_emit_render_pass_clears(list, true);
```

Comment:
How does this happen? If a clear is issued when that render target is not bound, shouldn't that get resolved immediately?

Reply:
The problem is that the app may bind a render target with `OMSetRenderTargets` that isn't defined in the pipeline. We don't know this until we actually bind a pipeline.

The second problem is that we cannot clear a depth-stencil attachment the aspects to clear are read-only in the render pass, which we also do not know in advance.

Reply:
Right, so binding a pipeline can invalidate the render target and carve out only a subset?

Reply:
Probably a good place to place a comment about that as well. "not used by pipeline" seems more directed than "not used by render pass".

Reply:
Actually, there might be another case where this is problematic.

What if attachments in a render pass have mismatched size, e.g. RT 0 is 64x64 and RT 1 is 128x128, we'll force renderArea of 64x64 in this case, but we actually wanted to clear the entire RT 1 as well. Images which are larger than the renderArea should probably be cleared separately as well.

Reply:
Seems rather esoteric though.